docs_src/design/adr/system-management/001-docker-image-guidelines.md

```diff
@@ -0,0 +1,44 @@
+# Docker image guidelines
+
+## Status
+
+Pending
+
+## Context
+
+When deploying the EdgeX Docker containers some security measures are recommended to ensure the integrity of the software stack.
+
+## Decision
+
+When deploying Docker images, the following flags should be set for heightened security.
+
+- To avoid escalation of privileges each docker container should use the `--security-opt=no-new-privileges` flag. More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#security-configuration)
+
+- To further prevent privilege escalation attacks the user should be set for the docker container using the `--user=<userid>` or `-u=<userid>` flag. More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#user)
+
+- To avoid a faulty or compromised containers from consuming excess amounts of the host of its resources `resource limits` should be set for each container. more details about `resource limits` can be found [here](https://docs.docker.com/config/containers/resource_constraints/)
+
+- To avoid attackers from writing data to the containers and modifying their files the `--read_only` flag should be set. More details about this flag can be found [here](https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir)
+
+> NOTE: exception
+    If a container is required to have write permission to function, then this flag will not work. For example, the vault requires write permission to write secrets. In this case the `--read_only` flag will not be used.
+
+> NOTE: alternatives
+    If writing non-persistent data is required (ex. a config file) then a temporary filesystem mount can be used to accomplish this goal while still enforcing `--read_only`. Mounting a `tmpfs` in Docker gives the container a temporary location in the host systems memory to modify files. This location will be removed once the container is stopped. More details about `tmpfs` can be found [here](https://docs.docker.com/storage/tmpfs/)
+
+for additional docker security rules and guidelines please check the Docker security [cheatsheet](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html)
+
+### Hadolint
+
+To help verify Docker standards are being used a static analysis tool should be used on all Dockerfiles. One such tool that is used by EdgeX is Hadolint.
+More information about hadolint can be found [here](https://github.com/hadolint/hadolint)
+
+NOTE: Visual Studio Code has a plugin for hadolint which may be used.
+
+## Consequences
+
+Create a more secure Docker environment
+
+## containers for the edge
+
+what's the standard for edgex containers
```

Comment:
Add a references section.  Include the following URLs:

- OWASP Docker Recommendations https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html
- CIS Docker Benchmark https://workbench.cisecurity.org/files/2433/download/2786 (registration required)
- Docker-compose reference https://docs.docker.com/compose/compose-file

Also provide cross-references for each recommendation above.